expfmt/text_parse.go

```diff
@@ -299,6 +299,25 @@ func (p *TextParser) startLabelName() stateFn {
 		p.parseError(fmt.Sprintf("expected '=' after label name, found %q", p.currentByte))
 		return nil
 	}
+	// Check for duplicate label names.
+	if len(p.currentMetric.Label) > 1 {
+		labelCount := make(map[string]int)
```

Comment:
You can do this more simply, as we don't need a count - only to detect a duplicate.

Reply:
I might be missing something, but how can I detect duplicates without keeping a count of the labels I have seen?

Reply:
If there is an entry in the map, then you know you have seen it. You only need a set, `map[string]struct{}`.